src/FamilyAttributeSet.php

```diff
@@ -0,0 +1,60 @@
+<?php
+namespace SnowIO\AkeneoDataModel;
+
+class FamilyAttributeSet implements \IteratorAggregate
+{
+    public static function empty(): self
+    {
+        return new self([]);
+    }
+
+    public function getAttribute(string $code): ?FamilyAttributeData
+    {
+        return $this->attributes[$code] ?? null;
+    }
+
+    public function filterByGroup(string $groupCode): self
+    {
+        return $this->filter(function (FamilyAttributeData $familyAttributeData) use ($groupCode) {
+            return $familyAttributeData->getGroup() === $groupCode;
+        });
+    }
+
+    public function filter(callable $predicate): self
+    {
+        $attributes = \array_filter($this->attributes, $predicate);
+        return new self($attributes);
+    }
+
+    public function isEmpty(): bool
+    {
+        return empty($this->attributes);
+    }
+
+    public static function fromJson(array $json): self
+    {
+        $attributes = [];
+        foreach ($json['attribute_groups'] as $attributeGroupJson) {
```

Comment:
If you only use the `'attribute_groups'` element of `$json` then the method's parameter should require the attribute groups JSON directly. I.e. this method should be changed to something like:

```php
foreach ($json as $attributeGroupJson) {

}
```

...and any code which calls `FamilyAttributeSet::fromJson()` should pass `$json['attribute_groups']` instead of `$json`.